in-app-wallet/typescript/src/in_app_wallet_examples.ts

```diff
@@ -77,11 +88,11 @@ const progTxnSplitGasless = (sender:string, sourceCoinId:string) => {
     );
     return txb;
 }
-  
+
 // The transaction block with gas context 
-const progTxnSplit = (sender:string, sourceCoinId:string) => {
+const progTxnSplit = (sender:string, sourceCoinId:string, gasCoinRef:SuiObjectRef) => {
```

Comment:
The original non-sponsored example uses the sourceCoinId as both the source to split from in the transaction, as well as the gas coin powering the txn. I thought this would be simpler as a tutorial. Did you explicitly separate these two out because this was too confusing?

Reply:
Yeah, I'm not sure if this was due to some changes but I wasn't able to get the code working with only the source coin set. I was able to use `txb.gas` but I thought it would be more clear to the end user what is going on with a specific coin used for gas. Also, on subsequent runs the `txb.gas` may not work depending on what source coin is used and when else is left over for use as gas.

Reply:
The only issue here would be if they are using a faucet, they won't be able to get two subsequent sui objects in a row due to faucet ratelimiting

Reply:
That's true, would you rather we use `txb.gas` then?

Reply:
Yeah I wasn't able to get that working before, I'll give it another try and see how it goes.

Reply:
I'm not really sure why I was having issues with that before, but it seems to be working now. I updated to just use the source coin.